File: gutenberg-downloader/DownloadBooks.java
import java.io.*;
import java.net.*;
import java.nio.file.*;
import java.util.concurrent.*;

public class DownloadBooks {
    public static void main(String[] args) {
        ExecutorService executor = Executors.newFixedThreadPool(5);
        for (int i = 1; i <= 100; i++) {
            final int id = i;
            executor.submit(() -> downloadBook(id));
        }
        executor.shutdown();
        try {
            executor.awaitTermination(1, TimeUnit.HOURS);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        System.out.println("Download complete.");
    }

    private static void downloadBook(int id) {
        String filename = "book_" + id + ".txt";
        Path path = Paths.get(filename);
        if (Files.exists(path)) return;
        String url = "https://www.gutenberg.org/cache/epub/" + id + "/pg" + id + ".txt";
        try {
            HttpURLConnection conn = (HttpURLConnection) new URL(url).openConnection();
            conn.setRequestMethod("GET");
            if (conn.getResponseCode() == 200) {
                try (BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
                     BufferedWriter out = Files.newBufferedWriter(path)) {
                    String line;
                    while ((line = in.readLine()) != null) {
                        out.write(line);
                        out.newLine();
                    }
                }
            }
        } catch (Exception e) {
            // silent
        }
    }
}

File: gutenberg-downloader/download_books.rs
use reqwest;
use tokio;

#[tokio::main]
async fn main() {
    let client = reqwest::Client::new();
    let mut handles = vec![];
    for i in 1..=100 {
        let client = client.clone();
        let handle = tokio::spawn(async move {
            download_book(client, i).await;
        });
        handles.push(handle);
    }
    for handle in handles {
        handle.await.unwrap();
    }
    println!("Download complete.");
}

async fn download_book(client: reqwest::Client, id: i32) {
    let filename = format!("book_{}.txt", id);
    if std::path::Path::new(&filename).exists() {
        return;
    }
    let url = format!("https://www.gutenberg.org/cache/epub/{}/pg{}.txt", id, id);
    match client.get(&url).send().await {
        Ok(resp) if resp.status().is_success() => {
            if let Ok(text) = resp.text().await {
                let _ = std::fs::write(&filename, text);
            }
        }
        _ => {}
    }
}